### PR TITLE
Add tracking to banner toggle events

### PR DESCRIPTION
**Pull request by EarthlingDavey**

### CODE REVIEW

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-events.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+        foreach ($posts->posts as $post_id) {
+            $all_events = get_metadata('post', $post_id, $this->event_details_field);
+            $all_post_events[$post_id] = $this->filterTrackEvents($all_events, $from, $to);
+        }
+
+        return $all_post_events;
+    }
+
+    // TODO: lifecycle policy, delete events older than x? Or keep only the most recent x events per post?
```
Comment (EarthlingDavey) [inline web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-events.php:165]: @wilson1000 fyi, I've added setters & getters for the tracking. These are the todos - I'll pick them up next.

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-admin.php [hunk 2]
```diff
@@ -324,7 +458,7 @@ public function menu(): void
             8
         );
 
-        add_action("load-$hook", [$this, 'pageLoad']);
+        // add_action("load-$hook", [$this, 'pageLoad']);
```
Comment (EmilyHazlehurst) [inline web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-admin.php:461]: @EarthlingDavey is this supposed to be commented out?

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+            fn ($banner) => [
+                'banner_active' => $banner['banner_active'],
+                'banner_content' => $banner['banner_content'],
+                'reference' => $banner['reference'],
+                'start_epoch' => strtotime($banner['start_date']),
+                'end_epoch' => $banner['end_date'] ? strtotime($banner['end_date']) : null
+            ],
+            $all_banners
+        );
+
```
Comment (EmilyHazlehurst) [inline web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php:61]: @EarthlingDavey not sure if he mentioned it but Damien noticed that the times were an hour behind yesterday - did you manage to sort that out?

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+        $email = [
+            'subject' => sprintf('Moj Intranet - Prior Party Banner Digest %d recent changes', $total_change_count),
+            'body' => $email_heading . implode('<br/>', $email_bodies)
+        ];
+
+        return $email;
+    }
+
+    // TODO: schedule task for digest emails.
```
Comment (EmilyHazlehurst) [inline web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php:313]: @EarthlingDavey is this TODO for a later date?

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-events.php [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+        }
+
+        $time = date($this->date_format_time, $event['time']);
+        $user = get_user_by('id', $event['user_id']);
+        $user_name = $user ? $user->display_name : 'Unknown';
+
+        return "User: $user_name,<br/> Action: {$event['action']},<br/> Time: $time";
+    }
+
+    // TODO: lifecycle policy, delete events older than x? Or keep only the most recent x events per post?
```
Comment (EmilyHazlehurst) [inline web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-events.php:214]: @EarthlingDavey is this TODO for a later date?

**Review by EmilyHazlehurst — APPROVED** [review]

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-admin.php (hunk 2 above)
Comment (EarthlingDavey) [reply web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-admin.php:461]: Thanks, this is to remind me to run through my decisions with Damien tomorrow :)

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php (hunk 3 above)
Comment (EarthlingDavey) [reply web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php:61]: Oh ! I don't know, will add it to the ticket.

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php (hunk 4 above)
Comment (EarthlingDavey) [reply web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php:313]: Cheers, I've got to come back to these.

File: web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-events.php (hunk 5 above)
Comment (EarthlingDavey) [reply web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-events.php:214]: Cheers, I've got to come back to these.

**Pull request merged**